grpc/execution/server.go

```diff
@@ -250,19 +214,26 @@ func (s *ExecutionServiceServerV1) ExecuteBlock(ctx context.Context, req *astria
 		return nil, status.Error(codes.PermissionDenied, "Cannot execute block until GetGenesisInfo && GetCommitmentState methods are called")
 	}
 
+	// the height that this block will be at
+	height := s.bc.CurrentBlock().Number.Uint64() + 1
+
+	fork := s.bc.Config().GetAstriaForks().GetForkAtHeight(height)
+
+	// this fork halts the chain
+	if fork.Halt {
+		return nil, status.Error(codes.FailedPrecondition, "Block cannot be created at halted fork")
+	}
```

Comment:
how does the chain un-halt?

Reply:
manual intervention

Reply:
could you link the doc for this? i'm unclear from the code as to how unhalting would work

Reply:
yes need to get some docs on this but basically the way it would work, such as the case of a sequencer network change, is you would pick some future height to halt at, update genesis to create a new named fork at that future height with `halt` set to `true`. Once the rollup reaches this height, it will stop. Then you do whatever manual steps you may need to do to prepare for the chain to migrate. e.g, updating conductor configs, taking any snapshots, etc. Then you update the geth genesis to set `halt` at this fork to `false` and set anything else you maybe did not know ahead of the halt, re-init and and start processing blocks.